app/controllers/application_controller.rb

```diff
@@ -5,6 +5,28 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    set :session_secret, "password_security"
+    enable :sessions
+
+  end
+
+  get '/' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'index'
+    end
+  end
+
+
+  helpers do
+    def logged_in?
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```